### ISSUE TITLE
Could not compile `ic-utils` during `cargo install icx-proxy`

### ISSUE THREAD

**Issue by zestified** [dfinity/icx-proxy#8]
Hi there! I'm getting the below error when trying to run `cargo install icx-proxy`.
```
   Compiling ic-agent v0.9.0
   Compiling ic-utils v0.7.0
error[E0277]: the trait bound `ic_agent::export::Principal: CandidType` is not satisfied
   --> /Users/phuongvu/.cargo/registry/src/github.com-1ecc6299db9ec823/ic-utils-0.7.0/src/interfaces/management_canister/builders.rs:251:42
    |
251 | impl<'agent, 'canister: 'agent, T: Sync> AsyncCall<(Principal,)>
    |                                          ^^^^^^^^^^^^^^^^^^^^^^^ the trait `CandidType` is not implemented for `ic_agent::export::Principal`
    |
    = note: required because of the requirements on the impl of `for<'de> ArgumentDecoder<'de>` for `(ic_agent::export::Principal,)`
note: required by a bound in `AsyncCall`
   --> /Users/phuongvu/.cargo/registry/src/github.com-1ecc6299db9ec823/ic-utils-0.7.0/src/call.rs:38:10
    |
36  | pub trait AsyncCall<Out>
    |           --------- required by a bound in this
37  | where
38  |     Out: for<'de> ArgumentDecoder<'de> + Send,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `AsyncCall`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ic-utils` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `icx-proxy v0.7.0`, intermediate artifacts can be found at `/var/folders/zj/f8bhtxm522vd_js96_y2w66c0000gn/T/cargo-installauI5dz`
```

**Comment by dubecker** [comment]
I am getting the same error. Did you find any solution yet?

**Comment by zestified** [comment]
@dubecker Sadly, I haven't.

**Comment by samlinux** [comment]
same problem on my side

**Comment by wewei** [comment]
Same problem on my side.
I'm on a mbp M1.

**Comment by ozwaldorf** [comment]
bumping this issue

**Comment by Daniel-Bloom-dfinity** [comment]
This should now be resolved with the release of `0.8.0`. Please re-open if you run into similar issues.

**Comment by Tyler-Gates** [comment]
Hey @Daniel-Bloom-dfinity, I am receiving this error in ic-utils v0.12.1

dfx v0.11.2
Ubuntu 20.04.4 LTS
```
Compiling ic-utils v0.12.1
error[E0277]: the trait bound `ic_agent::export::Principal: CandidType` is not satisfied
   --> /home/owner/.cargo/registry/src/github.com-1ecc6299db9ec823/ic-utils-0.12.1/src/interfaces/management_canister/builders.rs:251:42
    |
251 | impl<'agent, 'canister: 'agent, T: Sync> AsyncCall<(Principal,)>
    |                                          ^^^^^^^^^^^^^^^^^^^^^^^ the trait `CandidType` is not implemented for `ic_agent::export::Principal`
    |
    = note: required because of the requirements on the impl of `for<'de> ArgumentDecoder<'de>` for `(ic_agent::export::Principal,)`
note: required by a bound in `AsyncCall`
   --> /home/owner/.cargo/registry/src/github.com-1ecc6299db9ec823/ic-utils-0.12.1/src/call.rs:38:10
    |
36  | pub trait AsyncCall<Out>
    |           --------- required by a bound in this
37  | where
38  |     Out: for<'de> ArgumentDecoder<'de> + Send,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `AsyncCall`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ic-utils` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `icx-proxy v0.8.0`, intermediate artifacts can be found at `/tmp/cargo-installMQN1rO`
```

**Comment by hilljh82** [comment]
I am getting the same issue as well with the same version(s) as pointed out in @Tyler-Gates post.

**Comment by Daniel-Bloom-dfinity** [comment]
I've just published `icx-proxy v0.10.1` with updated dependencies, which should `cargo install` cleanly.
